Tests/Packet++Test/Tests/SomeIpTests.cpp

```diff
@@ -148,3 +149,161 @@ PTF_TEST_CASE(SomeIpCreationTest)
 	delete [] buffer1;
 	delete [] buffer2;
 }
+
+class SomeIpTeardown
+{
+public:
+	SomeIpTeardown(uint16_t someIpPort)
+	{
+		m_SomeIpPort = someIpPort;
+		pcpp::SomeIpLayer::addSomeIpPort(someIpPort);
+	}
+
+	~SomeIpTeardown()
+	{
+		pcpp::SomeIpLayer::removeSomeIpPort(m_SomeIpPort);
+	}
+private:
+	uint16_t m_SomeIpPort;
+};
+
+PTF_TEST_CASE(SomeIpTpParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	SomeIpTeardown someIpTeardown(16832);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/SomeIpTp1.dat");
+	READ_FILE_AND_CREATE_PACKET(2, "PacketExamples/SomeIpTp2.dat");
+
+	pcpp::Packet someIpTpPacket1(&rawPacket1);
+	pcpp::Packet someIpTpPacket2(&rawPacket2);
+
+	// Test SomeIpTpLayer::setTpFlag()
+	PTF_ASSERT_EQUAL((uint8_t)pcpp::SomeIpLayer::MsgType::TP_REQUEST_NO_RETURN, pcpp::SomeIpTpLayer::setTpFlag((uint8_t)pcpp::SomeIpLayer::MsgType::REQUEST_NO_RETURN));
+
+	// Test SOME/IP-TP start packet
+	PTF_ASSERT_TRUE(someIpTpPacket1.isPacketOfType(pcpp::SomeIP));
+	pcpp::SomeIpTpLayer *someIpTpLayer1 = someIpTpPacket1.getLayerOfType<pcpp::SomeIpTpLayer>();
+	PTF_ASSERT_NOT_NULL(someIpTpLayer1);
+	PTF_ASSERT_EQUAL(someIpTpLayer1->getHeaderLen(), 1412);
```

Comment:
Something in the math confuses me here:
The length field is 1404 + 2 bytes service ID + 2 bytes method ID = 1408.

How come the header len is 1412? 🤔 
I see that we calculate header len as:


I see that in Wireshark there are 4 extra bytes in this packet that are not included in this layer:

![image](https://user-images.githubusercontent.com/9059541/192715221-8a8a86b5-5de9-426d-9f75-e01c580a449a.png)

I think this is because we might miss something in `getHeaderLen()` calculation?

https://github.com/kolbex/PcapPlusPlus/blob/c97b2f6dfaf6209a7c5c7f43684f34de74fcf3cb/Packet%2B%2B/header/SomeIpLayer.h#L320

Maybe it should be:

```cpp
virtual size_t getHeaderLen() const { return sizeof(uint16_t) * 2 + getLengthField(); }
```